business_requirement_crm/wizard/crm_lead_convert_to_requirement.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Tecnativa - Victor M.M. Torres
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class CrmLeadConvertToRequirement(models.TransientModel):
+    _name = "crm.lead.convert.requirement"
+
+    lead_id = fields.Many2one(
+        'crm.lead',
+        string='Lead',
+        domain=[('type', '=', 'lead')])
+    business_requirement = fields.Html('Customer history')
+
+    @api.model
+    def default_get(self, fields):
+        result = super(CrmLeadConvertToRequirement, self).default_get(fields)
+        lead_id = self.env.context.get('active_id')
+        if lead_id:
+            name = self.env['crm.lead'].browse([lead_id]).name
+            result['lead_id'] = lead_id
+            result['business_requirement'] = name
+        return result
+
+    @api.multi
+    def action_lead_to_business_requirement(self):
+        self.ensure_one()
+        # get the lead to transform
+        lead = self.lead_id
+        customer_history = self.business_requirement
+        partner_id = lead.handle_partner_assignation()[lead.id]
+        business_requirement = ""
+        # create new business.requirement
+        if customer_history:
+            business_requirement = customer_history
+        elif lead.description:
+            business_requirement = lead.description
+        elif lead.name:
+            business_requirement = lead.name
+        vals = {
+            "name": lead.name,
+            "business_requirement": business_requirement,
+            "description": lead.description or lead.name,
+            "partner_id": partner_id,
+            "user_id": lead.user_id.id or lead.create_uid.id
+        }
+        requirement = self.env['business.requirement'].create(vals)
+        # move the mail thread
+        lead.message_change_thread(requirement)
+        # move attachments
+        attachments = self.env['ir.attachment'].search([
+            ('res_model', '=', 'crm.lead'),
+            ('res_id', '=', lead.id)])
+        attachments.write({
+            'res_model': 'project.task',
+            'res_id': requirement.id})
+        # archive the lead
+        lead.write({'active': False})
+        # return the action to go to the form view of the new BR
+        view = self.env.ref(
+            'business_requirement.view_business_requirement_form')
+        return {
+            'name': 'Requirement created',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'view_id': view.id,
+            'res_model': 'business.requirement',
+            'type': 'ir.actions.act_window',
+            'res_id': requirement.id,
+            'context': self.env.context
+        }
```

Comment:
You can open a form action more simple:
https://github.com/odoo/odoo/blob/b5a04fd29ff19839e788b2c383e3af74c883c54d/odoo/models.py#L1421